src/cider/nrepl/middleware/util/misc.clj

```diff
@@ -17,8 +17,15 @@
   (not (nil? (boot-fake-classpath))))
 
 (def java-api-version
-  (try (-> (System/getProperty "java.version") (str/split #"\.") second)
-       (catch Exception _ "7")))
+  (try
+    (let [java-ver (System/getProperty "java.version")
+          splited-ver (str/split java-ver #"\.")
+          old-ver (second splited-ver)
+          new-ver (first splited-ver)]
+      (if (< 1 (bigint new-ver))
```

Comment:
And I'd check if major is bigger than 1 `(> major 1)` - I think this makes the code easier to understand..